Apps/phvmray/vmray_connector.py

```diff
@@ -66,22 +123,25 @@ def _test_connectivity(self, _: Dict[str, Any]) -> bool:
         self.save_progress("Querying server to check connectivity")
 
         # Progress
-        self.save_progress(phantom.APP_PROG_CONNECTING_TO_ELLIPSES, server)
+        self.save_progress(phantom.APP_PROG_CONNECTING_TO_ELLIPSES.format(server))
 
         try:
             VMRay(server, api_key, not disable_cert)
         except VMRayRESTAPIError as exc:
-            self.set_status(phantom.APP_ERROR, VMRAY_ERR_SERVER_CONNECTION, exc)
-            self.append_to_message(VMRAY_ERR_CONNECTIVITY_TEST)
-            return self.get_status()
-
-        return self.set_status_save_progress(
-            phantom.APP_SUCCESS, VMRAY_SUCC_CONNECTIVITY_TEST
-        )
-
-    def _get_api(self) -> Tuple[bool, VMRay]:
+            err = self._get_error_message_from_exception(exc)
+            self.save_progress(VMRAY_ERR_CONNECTIVITY_TEST)
+            return action_result.set_status(phantom.APP_ERROR, VMRAY_ERR_SERVER_CONNECTION.format(err))
+        except Exception as e:
```

Comment:
Is there a reason why we catch a broad exception here and duplicate the code?

Reply:
Yes @jthom-vmray, we use this broad exception handling in order to avoid handle_exception from the connector if any other exception occurred from that code block. And, yes if you update the code accordingly to reduce the code duplication it would be good.

Reply:
Okay, thanks for the clarification. 👍
I'll adjust the exception handing.